source/common/upstream/cluster_manager_impl.cc

```diff
@@ -858,20 +858,15 @@ void ClusterManagerImpl::maybePreconnect(
   for (int i = 0; i < 3; ++i) {
     // Just as in ConnPoolImplBase::shouldCreateNewConnection, see if adding this one new connection
     // would put the cluster over desired capacity. If so, stop preconnecting.
-    if ((state.pending_streams_ + 1 + state.active_streams_) * peekahead_ratio <=
-        (state.connecting_stream_capacity_ + state.active_streams_)) {
+    if ((state.connecting_stream_capacity_ + state.active_streams_) >
+        (state.pending_streams_ + 1 + state.active_streams_) * peekahead_ratio) {
```

Comment:
I'm feeling very dense, but I'm having a really hard time with this logic. Two things:
1) We are mixing streams and connections. When you add one, do we need to be taking into account the number of streams per connection? It seems like there is math missing here but maybe I'm not understanding it (likely). (I have additional confusion about why we are adding +1 on this side of the equation vs. the other side, but maybe that will be more clear once we talk about connections vs. streams).
2) Is it possible to lift this logic out into a small helper shared with the same logic in `shouldCreateNewConnection` and add a bunch more comments? I think that would help me. (Same question about +1 in that function)